### PR TITLE
[FEATURE] - Custom Preset

### PR DESCRIPTION
**Pull request by andriiveremchuk**
Please have a look for any bad practices or/and redundant code. Thanks )

### CODE REVIEW

File: components/CustomPresetGrid.vue [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+      >
+        <div
+          v-for="(children, i) in childarea"
+          :key="children"
+          :class="'children' + i"
+          :style="{ gridArea: children }"
+        >
+          <button class="x-mark" @click="removeChild(i)">&#x2715;</button>
+        </div>
+        <div
```
Comment (andriytereschuk) [inline components/CustomPresetGrid.vue]: move to component

File: components/CustomPresetOptions.vue [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+<style lang="scss" scoped>
+$teal: #b388ff;
+$grey: #aaaaaa;
+
+aside {
+  margin: 60px;
+  font-size: 17px;
+  width: 300px;
+}
```
Comment (andriytereschuk) [inline components/CustomPresetOptions.vue]: clean up styles

File: components/CustomPresetOptions.vue [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+  opacity: 0.9;
+  font-weight: normal;
+  cursor: pointer;
+}
+@media screen and (max-width: 700px) {
+  aside {
+    width: 80vw;
+    margin: 100px 50px;
+  }
+}
```
Comment (andriytereschuk) [inline components/CustomPresetOptions.vue]: leave only what is needed

File: store/customPreset.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+export const state = () => ({
+  columns: 4,
+  rows: 2,
+  colArr: [],
+  rowArr: [],
+  childarea: [],
+  previewarea: null
+})
+
```
Comment (andriytereschuk) [inline store/customPreset.js]: move `getters` in the end

File: store/customPreset.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
+import { groupRepeatedUnits, createRepetition } from '../utils/helpers'
+
+export const state = () => ({
+  columns: 4,
+  rows: 2,
+  colArr: [],
+  rowArr: [],
```
Comment (andriytereschuk) [inline store/customPreset.js]: what is `childarea` and `preview` area?

File: store/customPreset.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    return createRepetition(unitGroups)
+  },
+  rowTemplate(state) {
+    const unitGroups = groupRepeatedUnits(state.rowArr)
+    return createRepetition(unitGroups)
+  },
+  divNum(state) {
+    return state.columns * state.rows
+  },
+  rowNumber(state) {
```
Comment (andriytereschuk) [inline store/customPreset.js]: consider return single item from the state as useless.
use `getters` for calculations, filtering, sorting, etc

File: store/customPreset.js (hunk 5 above)
Comment (andriiveremchuk) [reply store/customPreset.js]: I've renamed them to ***zones** and ***previewZone***

**Pull request merged**